src/types/error.ts

```diff
@@ -0,0 +1,55 @@
+export enum NetworkErrorTypes {
+  UnauthorizedError = 'UnauthorizedError', // 401
+  NotFoundError = 'NotFoundError', // 404
+  OrgNameConflictError = 'OrgNameConflictError', // 409
+  UnprocessableEntityError = 'UnprocessableEntity', // 422
+  ServerError = 'ServerError', // 500
+  GenericError = 'GenericError',
+}
+
+// 401 error
+export class UnauthorizedError extends Error {
+  constructor(message) {
+    super(message)
+    this.name = 'UnauthorizedError'
```

Comment:
I am not sure what this is being used for. I am guessing for response. If not then, I think we should use `xyz.constructor.name`. If this is being used in response, then we can just do `this.constructor.name` instead of hard-coding.

Reply:
Sorry - what would you propose changing this to? Won't referencing the `constructor` property return a reference to the constructor function itself, not the primitive value of name?

Reply:
```
this.name = this.constructor.name
```